### DISCUSSION TITLE
Connect to an other socket.io server

### DISCUSSION THREAD

**Discussion by ghost** [socketio/socket.io#4246] [Q&A] [answered]
Hello, I would like to know if there is a way to disconnect from a socket.io server and connect to an other with a new ip/domain without losing socket listeners ?

**Comment by darrachequesne** [comment, marked as answer]
Hi! You should be able to update the Manager `uri` attribute:

```js
import { io } from "socket.io-client";

const socket = io("https://my-domain-a.com");

socket.io.uri = "https://my-domain-b.com";

// force reconnection
socket.disconnect().connect();
```

The TypeScript compiler might complain though, as it's a `private readonly` attribute.

**Reply by ghost** [reply]
Thanks !

**Reply by mikejpeters** [reply]
This is super helpful, I just want to note for anyone else doing this, I don't think setting a new `uri` will update the namespace. 

In my case I was initializing `io` without passing any uri, which caused it to use the currrent URL, and incorrectly use a namespace like `//capacitor://localhost` even after I updated the `uri` to something like `https://example.com` using this method. I ended up initializing it with a dummy uri so it doesn't try to use the current URL and a non-default namespace.
